rinja_derive/src/generator/node.rs

```diff
@@ -616,14 +617,28 @@ impl<'a> Generator<'a, '_> {
                     call.span(),
                 )
             })?;
-            (def, mctx)
+            (*def, mctx)
         } else {
             let def = ctx.macros.get(name).ok_or_else(|| {
                 ctx.generate_error(format_args!("macro {name:?} not found"), call.span())
             })?;
-            (def, ctx)
+            (*def, ctx)
         };
 
+        if self.seen_macros.iter().any(|(s, _)| std::ptr::eq(*s, def)) {
```

Comment:
I have a few cases where this is problematic:

```jinja
{% macro some_macro(add) %}
    {% if add < 10 %}
        {{ add }}
        {% call next::some_macro(add + 1) %}
    {% endif %}
{% endmacro %}
```

I think we should instead count the recursion instead of forbid it.

Reply:
Your example causes a stackoverflow in the current implementation. The condition is only checked at runtime, and the macro body is always entered. No counting needed, once we are in a loop, we'll always be in a loop.

Reply:
Oh I see. Hum... Should we transform macros into functions?

Reply:
I think that might be a good idea for a follow up PR. When we encounter a loop, we can transform the call into a function call instead of returning an error. I would not always turn macro into functions, though.

Reply:
Fair enough.